src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,216 @@
+            getAllByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCar(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get list of car by driver id " + driverId,
+                    throwable);
+        }
+        setDrivers(cars);
+        return cars;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        Manufacturer manufacturer = new Manufacturer(resultSet.getString("name"),
+                resultSet.getString("country"));
+        manufacturer.setId(manufacturerId);
+        Long carId = resultSet.getObject("id", Long.class);
+        Car car = new Car(resultSet.getString("model"), manufacturer);
+        car.setId(carId);
+        return car;
+    }
+
+    private List<Driver> getDriverListByCarId(Long carId) {
+        String query = "SELECT d.id, d.name, d.license_number "
+                + "FROM cars c JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "JOIN drivers d ON cd.driver_id = d.id "
+                + "WHERE c.id = ? AND c.deleted = FALSE;";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversByCarStatement
+                        = connection.prepareStatement(query)) {
+            getDriversByCarStatement.setLong(1, carId);
+            ResultSet resultSet = getDriversByCarStatement.executeQuery();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                String name = resultSet.getString("name");
+                String licenseNumber = resultSet.getString("license_number");
+                Driver driver = new Driver(name, licenseNumber);
+                driver.setId(id);
+                drivers.add(driver);
+            }
+            return drivers;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get list of drivers by car id" + carId,
+                    throwable);
+        }
+    }
+
+    private boolean insertDrivers(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUE (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertDriversStatement
+                        = connection.prepareStatement(query)) {
+            insertDriversStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                insertDriversStatement.setLong(2, driver.getId());
+                insertDriversStatement.executeUpdate();
+            }
+            return insertDriversStatement.executeUpdate() > 0;
```

Comment:
It seems like in return statement the  executeUpdate() method is called again and executes the same query again. may be its better to put counter inside the loop and return counter == car.getDrivers().size()?